### PR TITLE
[RDBMS] `az mysql flexible-server restore/geo-restore`: Add parameters to enhance PITR

### PR DESCRIPTION
**Pull request by honghr**
**Related command**

- az mysql flexible-server restore --sku_name --tier --storage-size --storage-auto-grow --backup_retention --geo-redundant-backup
- az mysql flexible-server geo-restore --sku_name --tier --storage-size --storage-auto-grow --backup_retention --geo-redundant-backup

**Description**

We are enhancing the existing PITR experience by providing additional flexibility to the user to change storage+compute setting during restore and geo restore.

**Testing Guide**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Comment by honghr** [comment]
Hi @evelyn-ys , What do I need to do next to get the merge?

**Comment by evelyn-ys** [comment]
@honghr Please make sure all CI can pass

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -525,15 +525,21 @@ def _flexible_server_params(command_group):
             c.argument('subnet', arg_type=subnet_arg_type)
             c.argument('subnet_address_prefix', arg_type=subnet_address_prefix_arg_type)
             c.argument('private_dns_zone_arguments', private_dns_zone_arguments_arg_type)
-            c.argument('geo_redundant_backup', default='Disabled', arg_type=geo_redundant_backup_arg_type)
```
Comment (Ruiion) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:528]: why we need to remove this argument and add it another place

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**